### PR TITLE
AM-002 [BE] Get All and Get By Id Assets from the DB - Completed

### PR DESCRIPTION
**Pull request by DTA-Wijesekara**
AM-002 [BE] Get All and Get By Id Assets from the DB

### CODE REVIEW

File: AssetManagementSystem/Controllers/AssetsController.cs [hunk 1]
```diff
@@ -29,5 +30,23 @@ public async Task<IActionResult> AddAsset([FromBody] AssetDto assetDto)
             return Ok(addedAsset);
         }
 
+        [HttpGet]
+        public async Task<IActionResult> GetallAsset()
```
Comment (HasangaLakdinu) [inline AssetManagementSystem/Controllers/AssetsController.cs:34]: GetAllAssets

File: AssetManagementSystem/Controllers/AssetsController.cs [hunk 2]
```diff
@@ -29,5 +30,23 @@ public async Task<IActionResult> AddAsset([FromBody] AssetDto assetDto)
...
+        {
+            var AssetsAll = await _assetService.GetAllAssetsAsync();
```
Comment (HasangaLakdinu) [inline AssetManagementSystem/Controllers/AssetsController.cs:36]: var AllAssets

File: AssetManagementSystem/Controllers/AssetsController.cs [hunk 3]
```diff
@@ -29,5 +30,23 @@ public async Task<IActionResult> AddAsset([FromBody] AssetDto assetDto)
...
+            return Ok(AssetsAll);
+        }
+
+        [HttpGet]
+        [Route("{id:int}")]
+        public async Task<IActionResult> GetAssetById([FromRoute]int id)
+        {
+            var AssetOne = await _assetService.GetAssetByIdAsync(id);
```
Comment (HasangaLakdinu) [inline AssetManagementSystem/Controllers/AssetsController.cs:44]: AssetOne---> SelectedAsset

File: AssetManagementSystem/Services/AssetService.cs [hunk 4]
```diff
@@ -34,5 +36,17 @@ public async Task<Asset> AddAssetAsync(AssetDto newAsset)
             await _dbContext.SaveChangesAsync();
             return NewAsset;
         }
+
+        public async Task<List<Asset>> GetAllAssetsAsync()
+        {
+            var Assetz = await _dbContext.Asset.ToListAsync();
```
Comment (HasangaLakdinu) [inline AssetManagementSystem/Services/AssetService.cs:42]: Assets

File: AssetManagementSystem/Services/AssetService.cs [hunk 5]
```diff
@@ -34,5 +36,17 @@ public async Task<Asset> AddAssetAsync(AssetDto newAsset)
...
+            return Assetz;
+        }
+
+        public async Task<Asset> GetAssetByIdAsync(int id)
+        {
+            var Assetz = await _dbContext.Asset.FirstOrDefaultAsync(x => x.Id == id);
```
Comment (HasangaLakdinu) [inline AssetManagementSystem/Services/AssetService.cs:48]: SelectedAsset

File: AssetManagementSystem/Controllers/AssetsController.cs (hunk 1 above)
Comment (DTA-Wijesekara) [reply AssetManagementSystem/Controllers/AssetsController.cs]: done

File: AssetManagementSystem/Controllers/AssetsController.cs (hunk 2 above)
Comment (DTA-Wijesekara) [reply AssetManagementSystem/Controllers/AssetsController.cs]: done

File: AssetManagementSystem/Controllers/AssetsController.cs (hunk 3 above)
Comment (DTA-Wijesekara) [reply AssetManagementSystem/Controllers/AssetsController.cs]: done

File: AssetManagementSystem/Services/AssetService.cs (hunk 4 above)
Comment (DTA-Wijesekara) [reply AssetManagementSystem/Services/AssetService.cs]: done

File: AssetManagementSystem/Services/AssetService.cs (hunk 5 above)
Comment (DTA-Wijesekara) [reply AssetManagementSystem/Services/AssetService.cs]: done

**Pull request merged**